pkg/reconciler/taskrun/taskrun_test.go

```diff
@@ -255,22 +261,55 @@ func getRunName(tr *v1beta1.TaskRun) string {
 	return strings.Join([]string{tr.Namespace, tr.Name}, "/")
 }
 
+func ensureConfigurationConfigMapsExist(d *test.Data) {
+	var defaultsExists, featureFlagsExists bool
+	for _, cm := range d.ConfigMaps {
+		if cm.Name == config.GetDefaultsConfigName() {
+			defaultsExists = true
+		}
+		if cm.Name == config.GetFeatureFlagsConfigName() {
+			featureFlagsExists = true
+		}
+	}
+	if !defaultsExists {
+		d.ConfigMaps = append(d.ConfigMaps, &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{Name: config.GetDefaultsConfigName(), Namespace: system.GetNamespace()},
+			Data:       map[string]string{},
+		})
+	}
+	if !featureFlagsExists {
+		d.ConfigMaps = append(d.ConfigMaps, &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{Name: config.GetFeatureFlagsConfigName(), Namespace: system.GetNamespace()},
+			Data:       map[string]string{},
+		})
+	}
+}
+
 // getTaskRunController returns an instance of the TaskRun controller/reconciler that has been seeded with
 // d, where d represents the state of the system (existing resources) needed for the test.
-func getTaskRunController(t *testing.T, d test.Data) (test.Assets, func()) {
+func getTaskRunController(t *testing.T, d test.Data) (test.Assets, func(), chan<- struct{}) {
 	//unregisterMetrics()
 	ctx, _ := ttesting.SetupFakeContext(t)
 	ctx, cancel := context.WithCancel(ctx)
 	cloudEventClientBehaviour := cloudevent.FakeClientBehaviour{
 		SendSuccessfully: true,
 	}
 	ctx = cloudevent.WithClient(ctx, &cloudEventClientBehaviour)
+	ensureConfigurationConfigMapsExist(&d)
 	c, _ := test.SeedTestData(t, ctx, d)
 	configMapWatcher := configmap.NewInformedWatcher(c.Kube, system.GetNamespace())
+
+	controller := NewController(namespace, images)(ctx, configMapWatcher)
+
+	stopCh := make(chan struct{})
+	if err := configMapWatcher.Start(stopCh); err != nil {
+		t.Fatalf("error starting configmap watcher: %v", err)
+	}
+
 	return test.Assets{
-		Controller: NewController(namespace, images)(ctx, configMapWatcher),
+		Controller: controller,
 		Clients:    c,
-	}, cancel
+	}, cancel, stopCh
```

Comment:
Thank you for the review. Fixed. I will take a look and submit a PR if there is a way to avoid dealing with the channel altogether, i.e. letting `SetupFakeContext` handle it.